internal/service/federatedsettingsidentityprovider/data_source_federated_settings_identity_provider.go

```diff
@@ -281,11 +281,11 @@ func dataSourceMongoDBAtlasFederatedSettingsIdentityProviderRead(ctx context.Con
 		return diag.FromErr(fmt.Errorf("error setting `status` for federatedSettings IdentityProviders: %s", err))
 	}
 
-	if err := d.Set("id", federatedSettingsIdentityProvider.Id); err != nil {
-		return diag.FromErr(fmt.Errorf("error setting `id` for federatedSettings IdentityProviders: %s", err))
+	if err := d.Set("idp_id", federatedSettingsIdentityProvider.Id); err != nil {
+		return diag.FromErr(fmt.Errorf("error setting `idp_id` for federatedSettings IdentityProviders: %s", err))
 	}
 
-	d.SetId(federatedSettingsIdentityProvider.OktaIdpId)
+	d.SetId(federatedSettingsIdentityProvider.Id)
```

Comment:
same question here... wouldn't this be a breaking change because the value in the TF state change?

Reply:
This part of the code will only be reached if/when the user has re imported the resource with the new 24-hex characters ID. In the case that the user has not followed yet the upgrade guide, it will reach [this point](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1842/files#diff-3b955d8e92a25faa6dd8d3e000e684fe134f3484171d07860837cc78ae724bfaR352)  inside `oldSDKDSRead` [method](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1842/files#diff-3b955d8e92a25faa6dd8d3e000e684fe134f3484171d07860837cc78ae724bfaR293) and no breaking change will happen.

Reply:
Actually I believe this is a good catch because this is for the single data source implementation. When the user updates his config with the new format in `identity_provider_id` argument, I believe we should still preserve `OktaIdpId` as the id value of the data source in case the user is referencing the id of this data source. He will still be able to access the new format in the `idp_id` attribute.